pkgs/desktops/gnome-3/core/baobab/default.nix

```diff
@@ -1,30 +1,54 @@
-{ stdenv, gettext, fetchurl, vala, desktop-file-utils
-, meson, ninja, pkgconfig, gtk3, glib, libxml2
-, wrapGAppsHook, itstool, gnome3 }:
-
+{ stdenv, lib, gettext, fetchurl, vala, desktop-file-utils
+, meson , ninja , pkgconfig , gtk3 , glib , libxml2
+, wrapGAppsHook , itstool , gnome3
+, gnused
+}:
 let
   pname = "baobab";
   version = "3.34.0";
 in stdenv.mkDerivation rec {
+
   name = "${pname}-${version}";
 
-  src = fetchurl {
-    url = "mirror://gnome/sources/${pname}/${stdenv.lib.versions.majorMinor version}/${name}.tar.xz";
-    sha256 = "01w6m5ylyqs4wslpjn1hg6n6ynwh3rghksak0qs8r9m6dm3dkss6";
-  };
+  srcs = [ (fetchurl {
+      url = "mirror://gnome/sources/${pname}/${lib.versions.majorMinor version}/${name}.tar.xz";
+      sha256 = "01w6m5ylyqs4wslpjn1hg6n6ynwh3rghksak0qs8r9m6dm3dkss6";
+    })
+    ./baobab-icons.tar.gz
+  ];
+
+  sourceRoot = "${name}";
 
   nativeBuildInputs = [ meson ninja pkgconfig vala gettext itstool libxml2 desktop-file-utils wrapGAppsHook ];
-  buildInputs = [ gtk3 glib gnome3.adwaita-icon-theme ];
+  buildInputs = [ gnused gtk3 glib gnome3.adwaita-icon-theme ];
 
   doCheck = true;
 
+  postPatch = ''
+    cp -r ../icons .
+  '';
+
+  # Deletes traductions for Icon Entry in .desktop
+  # And deletes the NotShowIn=KDE thing.
+  postBuild = ''
+    sed -i "/^Icon\[.*\].*$/d" data/org.gnome.baobab.desktop
```

Comment:
What is wrong with translations? Translators are explicitly told not to translate the icon names. It would only be used for RTL icon if applicable.